File: src/main/java/no/nav/aap/api/søknad/tokenx/TokenXConfigMatcher.java
package no.nav.aap.api.søknad.tokenx;

import no.nav.security.token.support.client.core.ClientProperties;
import no.nav.security.token.support.client.spring.ClientConfigurationProperties;

import java.net.URI;
import java.util.Optional;

 interface TokenXConfigMatcher {
    Optional<ClientProperties> findProperties(ClientConfigurationProperties configs, URI uri);
}

File: src/main/kotlin/no/nav/aap/api/rest/tokenx/TokenXConfigMatcher.kt
package no.nav.aap.api.rest.tokenx

import no.nav.security.token.support.client.core.ClientProperties
import no.nav.security.token.support.client.spring.ClientConfigurationProperties
import java.net.URI

@FunctionalInterface
interface TokenXConfigMatcher {
    fun findProperties(configs: ClientConfigurationProperties, uri: URI): ClientProperties?
}